cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -257,20 +248,77 @@ private String getSchemaCompatibility(
     }
   }
 
+  private Pair<String, String> getSubjectCompatibilityWithGlobalFallBack(
+      String environmentVal,
+      String topicName,
+      KafkaSupportedProtocol protocol,
+      String clusterIdentification) {
+    try {
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String compatibility =
+          getSubjectSchemaCompatibility(environmentVal, topicName, protocol, clusterIdentification);
+
+      if (!compatibility.equals(SCHEMA_COMPATIBILITY_NOT_SET)) {
+        return Pair.of(SUBJECT, compatibility);
+      }
+
+      String suffixUrl = environmentVal + "/config/";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol);
+
+      ResponseEntity<Map<String, String>> responseList =
+          getSubjectSchemaCompatibility(
+              reqDetails, new HashMap<>(), createSchmeaRegistryRequest(clusterIdentification));
+      log.info("Schema compatibility " + responseList);
+      return Pair.of(GLOBAL, responseList.getBody().get("compatibilityLevel"));
```

Comment:
possibility for npe

Reply:
Deleted method as no longer required for change in logic.